Actions/ReadPowerPlatformSettings/ReadPowerPlatformSettings.ps1

```diff
@@ -5,22 +5,31 @@ param(
     [string] $environmentName
 )
 
+$envName = $environmentName.Split(' ')[0]
+
+# Read the deployment settings
 $deploymentEnvironments = $deploymentEnvironmentsJson | ConvertFrom-Json
 $deploymentSettings = $deploymentEnvironments."$environmentName"
-$envName = $environmentName.Split(' ')[0]
-$secrets = $env:Secrets | ConvertFrom-Json
 
 foreach($property in 'ppEnvironmentUrl','companyId','environmentName') {
     if ($deploymentSettings."$property") {
         Write-Host "Setting $property"
         Add-Content -Encoding utf8 -Path $env:GITHUB_OUTPUT -Value "$property=$($deploymentSettings."$property")"
     }
     else {
-        Write-Host "::ERROR::DeployTo$envName setting must contain '$property' property"
-        exit 1
+        # Write-Host "::ERROR::DeployTo$envName setting must contain '$property' property"
+        throw "::ERROR::$envName setting must contain '$property' property"
```

Comment:
don't include ::error:: when you throw - that gets added by the try catch handler in action.yaml